synapse/storage/databases/main/devices.py

```diff
@@ -1078,16 +1078,18 @@ async def store_device(
             return False
 
         try:
-            inserted = await self.db_pool.simple_insert(
+            inserted = await self.db_pool.simple_upsert(
                 "devices",
-                values={
+                keyvalues={
                     "user_id": user_id,
                     "device_id": device_id,
+                },
+                values={},
+                insertion_values={
                     "display_name": initial_device_display_name,
                     "hidden": False,
                 },
                 desc="store_device",
-                or_ignore=True,
             )
             if not inserted:
```

Comment:
This block will now never run for postgresql, this seems like a change in behavior, but I'm not sure if it matters or not.

Reply:
It should? The idea is that `simple_upsert` will return False if we didn't insert or update a new row, and since `values={}` we'll not update rows and so we'll hit this condition if we didn't insert a new entry?

Reply:
I was thinking `inserted` would also be true, but yeah I see if `values` is `{}` then this isn't true.